### PR TITLE
tell cargo about all our custom cfgs

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

### CODE REVIEW

File: hil-test/build.rs [hunk 1]
```diff
@@ -34,6 +34,8 @@ fn main() -> Result<(), Box<dyn Error>> {
     let chip = Chip::from_str(device_name)?;
     let config = Config::for_chip(&chip);
 
+    esp_metadata::define_all_possible_symbols();
```
Comment (bugadani) [inline hil-test/build.rs:37]: I guess this could happen inside `define_symbols` so we will never forget to call it?

Comment (MabezDev) [reply hil-test/build.rs]: Good idea! Thanks!

**Review by bugadani — APPROVED** [review]
<3

**Comment by jessebraham** [comment]
Why is the `clic` module being removed here?

**Comment by MabezDev** [comment]
> Why is the clic module being removed here?

It's currently not an exercised code path, it should have been removed with the P4 support. We can always add it back when we add P4 support.

**Pull request merged**